ssh2/sftp_handle.pyx

```diff
@@ -270,7 +270,12 @@ cdef class SFTPHandle:
         cdef char *cbuf = buf
         cdef ssize_t rc
         with nogil:
-            rc = c_sftp.libssh2_sftp_write(self._handle, cbuf, _size)
+            while _size > 0:
```

Comment:
With this change if `_size` is 0 `rc` will be undefined and will cause a crash. Can skip call to write and just return 0 in that case.